### ISSUE TITLE
result not getting checked in post request

### ISSUE THREAD

**Issue by kunwarsingh1** [code100x/neet-results-2024#2]
![Screenshot 2024-06-12 230208](https://github.com/code100x/neet-results-2024/assets/122791808/191b41f5-4eea-47c2-a108-0922115e6a8f)

in the above ss we can see that in the post request we are fetching for the result but not checking if the worker's result is same as that of the original result. Is is getting checked at any point?

**Comment by devsargam** [comment]
Yeah, that is checked here

https://github.com/code100x/neet-results-2024/blob/main/src/worker/sendRequest.ts#L52

**Comment by kunwarsingh1** [comment]
I don't think that line of code is checking if the response created by the worker is legit or not

**Comment by devsargam** [comment]
It does

**Comment by Ryneex** [comment]
@devsargam maybe he was asking about validation of results returned by workers.

**Comment by kunwarsingh1** [comment]
@Ryneex yes

**Comment by devsargam** [comment]
Oh yeah good catch!!! @kunwarsingh1 @Ryneex 
But should not be much problem as we get valid dob and application number

**Comment by kunwarsingh1** [comment]
we are getting valid dob and application number but the problem will arise if a hacker makes a false post request using maybe postman or so giving invalid data about that particular application number and dob. On the database side it will be an invalid entry and thus the idea of decentralized backend will be compromised

**Comment by devsargam** [comment]
@Ryneex he means, a hacker gives correct application number and dob but gives incorrect marks, air and name. The database will still store the faulty marks, air and names

**Comment by Ryneex** [comment]
@devsargam looks like it got compromised, he should've trusted the data returned by result instead of req

**Comment by devsargam** [comment]
We are fine lol, with only 9k entires with valid dob and application number it should not be much hassle to find all the correct marks 👍

**Comment by kunwarsingh1** [comment]
so basically the await function is there for no reason and is just increasing the time of the request and if this is the case then it will cost a small amount of money to @hkirat which I don't think is good.
